### PR TITLE
cmake: fix check for memfd_create

### PR DESCRIPTION
**Pull request by bkuhls**
## Description

memfd_create is disabled on uClibc:
https://cgit.uclibc-ng.org/cgi/cgit/uclibc-ng.git/commit/?id=0bbc6a4bf54197ec5dab874750d125cbd5b9e877

Instead of checking for the header file we need to check for
memfd_create(), fixes build error

xbmc/utils/UDMABufferObject.cpp:94:13: error:
 ‘memfd_create’ was not declared in this scope

for this kodi config:

-DCORE_PLATFORM_NAME=wayland
-DWAYLAND_RENDER_SYSTEM=gl

-- Core system type: linux
-- Platform: wayland

## Motivation and Context

Fix build with wayland and uClibc

## How Has This Been Tested?

Build-tested using buildroot.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document

### CODE REVIEW

File: cmake/scripts/linux/ArchSetup.cmake [hunk 1]
```diff
@@ -84,11 +84,11 @@ endif()
 include(LDGOLD)
 
 include(CheckIncludeFiles)
-check_include_files("linux/memfd.h" HAVE_LINUX_MEMFD)
+check_function_exists(memfd_create HAVE_LINUX_MEMFD)
```
Comment (lrusak) [inline cmake/scripts/linux/ArchSetup.cmake:87]: This won't work IIRC, you need to supply the header and likely the library to link againts
`CMAKE_REQUIRED_INCLUDES` and `CMAKE_REQUIRED_LIBRARIES`

see: https://cmake.org/cmake/help/v3.8/module/CheckFunctionExists.html

Comment (bkuhls) [reply cmake/scripts/linux/ArchSetup.cmake]: > see: https://cmake.org/cmake/help/v3.8/module/CheckFunctionExists.html

Cmake recommends to use CheckSymbolExists instead, sent new iteration of the patch.

**Comment by wsnipex** [comment]
please move the whole hunk further down to the other check_symbol_exists check for MKOSTEMP

**Comment by bkuhls** [comment]
> please move the whole hunk further down to the other check_symbol_exists check for MKOSTEMP

done

**Review by wsnipex — APPROVED** [review]

**Pull request merged**